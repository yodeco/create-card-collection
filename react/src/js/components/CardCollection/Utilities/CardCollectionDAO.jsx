/* eslint-disable */
export default class CardCollectionDAO {
    static SORT_BY_NAME = 'name';
    static SORT_BY_POPULAR = 'popular';

    /**
     * Creates an instance of CardCollectionDAO.
     * @param {object} [{
     *         endpoint = '',
     *         pageType = 'collection',
     *         sort = CardCollectionDAO.SORT_BY_POPULAR,
     *         results = 10,
     *         page = 1,
     *     }={}]
     * @memberof CardCollectionDAO
     */
    constructor({
        endpoint = '',
        sort = CardCollectionDAO.SORT_BY_POPULAR,
        activeFilters = [],
        results = 10,
        page = 1,
    } = {}) {
        this.endpoint = endpoint;
        this.activeFilters = activeFilters;
        this.sort = sort;
        this.results = results;
        this.page = page;
    }

    static shuffle(array) {
        let currentIndex = array.length,
            temporaryValue,
            randomIndex;

        // While there remain elements to shuffle...
        while (currentIndex !== 0) {
            // Pick a remaining element...
            randomIndex = Math.floor(Math.random() * currentIndex);
            currentIndex -= 1;

            // And swap it with the current element.
            temporaryValue = array[currentIndex];
            array[currentIndex] = array[randomIndex];
            array[randomIndex] = temporaryValue;
        }

        return array;
    }

    /**
     * Get JSON data from the server with provided parameters
     *
     * @returns {Promise} Promise Object with server json data
     * @memberof CardCollectionDAO
     */
    getData() {
        let fetchStr = `${this.endpoint}.collection.json/sort-${this.sort}/`;
        if (this.activeFilters.length !== 0) {
            const filterStr = this.activeFilters.join('/');
            fetchStr += `${filterStr}/`;
        }
        fetchStr += `results-${this.results}.${this.page}.json`;
        const responsePromise = fetch(fetchStr, { credentials: 'same-origin' })
            .then(response => response.json());
        return responsePromise;
        // const cardCollectionMock = window.cardCollectionMock;
        // CardCollectionDAO.shuffle(cardCollectionMock.cards);
        // return new Promise(((resolve) => {
        //     setTimeout(() => {
        //         resolve(cardCollectionMock);
        //     }, 1000);
        // }));
    }

    /**
     * Set the sort order of the collection.
     * Current valid values are the static values SORT_BY_NAME or SORT_BY_POPULAR
     *
     * @param {string} sort
     * @returns {CardCollectionDAO} <code>this</code>
     * @memberof CardCollectionDAO
     */
    setSort(sort) {
        if (sort === CardCollectionDAO.SORT_BY_NAME ||
            sort === CardCollectionDAO.SORT_BY_POPULAR) {
            this.sort = sort;
        } else {
            // If sort is not a valid value, sort by popular
            this.sort = CardCollectionDAO.SORT_BY_POPULAR;
        }
        return this;
    }

    /**
     * Set the number of results to be returned by the fetch
     *
     * @param {int} results
     * @returns {CardCollectionDAO} <code>this</code>
     * @memberof CardCollectionDAO
     */
    setResults(results) {
        const newResults = parseInt(results, 10);
        if (Number.isNaN(newResults)) {
            this.results = 10;
        } else {
            this.results = results;
        }
        return this;
    }

    /**
     * Set which page of results to fetch
     *
     * @param {int} page
     * @returns {CardCollectionDAO} <code>this</code>
     * @memberof CardCollectionDAO
     */
    setPage(page) {
        const newPage = parseInt(page, 10);
        if (Number.isNaN(newPage)) {
            this.page = 1;
        } else {
            this.page = page;
        }
        return this;
    }

    /**
     * Get the next page of results data
     *
     * @returns {Promise} Promise Object with server json data
     * @memberof CardCollectionDAO
     */
    nextPage() {
        this.page = this.page + 1;
        return this.getData();
    }
}

export const a = 5;
