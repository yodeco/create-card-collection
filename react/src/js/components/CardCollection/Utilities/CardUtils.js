import * as _ from 'lodash';


export function shouldShowArticleUrl(props) {
    const articleUrl = _.get(props, 'article.url', '');
    return articleUrl !== '';
}

export function shouldShowArticleDescription(props) {
    const articleDescription = _.get(props, 'article.description', '');
    return articleDescription !== '';
}


export function showFeaturedCardCta(props) {
    const ctaButtonText = _.get(props, 'article.ctaButtonText', '');
    const ctaButtonLink = _.get(props, 'article.ctaButtonLink', '');
    return ctaButtonText !== '' && ctaButtonLink !== '';
}

/*
 * Get the page name sans domain and html
 */
export function getPageName(articleUrl) {
    const pageNameIndex = articleUrl.lastIndexOf('/');
    const pageNamePlusExt = articleUrl.substring(pageNameIndex + 1);
    return pageNamePlusExt.split('.')[0];
}

export function shouldShowTagLabel(props) {
    const tagTitle = _.get(props, 'tag.title', '');
    return tagTitle !== '';
}

export function shouldShowVideoUrl(props) {
    const videoUrl = _.get(props, 'video.url', '');
    return videoUrl !== '';
}


export function getByAuthorText(author, byText) {
    let authorStr = author || '';
    if (authorStr.length && byText && byText.includes('{authorName}')) {
        authorStr = byText.replace('{authorName}', authorStr);
    }
    return authorStr;
}


export function setSubscriptionCardProps(props, obj) {
    /* image properties */
    obj.imageSize = _.get(props, 'styles.imageSize', '');
    obj.backgroundImageUrl = _.get(props, 'background', '');
    obj.imageHorizontalAlignment = _.get(props, 'styles.imageHorizontalAlignment', '50%');
    obj.imageVerticalAlignment = _.get(props, 'styles.imageVerticalAlignment', '50%');
    obj.imageAlt = _.get(props, 'imageAlt', '');
    obj.imageSize = obj.imageSize === 'custom' ? obj.customImageSize : obj.imageSize;

    /* title card styles */
    obj.backgroundColor = _.get(props, 'styles.backgroundColor', '');
    obj.descriptionColor = _.get(props, 'styles.descriptionColor', '');
    obj.titleColor = _.get(props, 'styles.titleColor', '');
    obj.thankYouTitleColor = _.get(props, 'styles.thankYouView.titleColor', '');
    obj.thankYouDescriptionColor = _.get(props, 'styles.thankYouView.descriptionColor', '');

    /* article properties */
    obj.adobeCampaignServiceName = _.get(props, 'article.adobeCampaignServiceName', '');
    obj.consentNoticeText = _.get(props, 'article.consentNoticeText', '');
    obj.ctaButtonText = _.get(props, 'article.ctaButtonText', '');
    obj.articleDescription = _.get(props, 'article.description', '');
    obj.emailPlaceholderText = _.get(props, 'article.emailPlaceholderText', '');
    obj.invalidEmailText = _.get(props, 'article.invalidEmailText', '');
    obj.namePlaceholderText = _.get(props, 'article.namePlaceholderText', '');
    obj.requiredText = _.get(props, 'article.requiredText', '');
    obj.serverErrorText = _.get(props, 'article.serverErrorText', '');
    obj.subscriptionAPI = _.get(props, 'article.subscriptionAPI', '');
    if (obj.subscriptionAPI === '') {
        obj.subscriptionAPI = 'https://www.adobe.com/api2/subscribe_v1';
    }
    obj.subscriptionCountry = _.get(props, 'article.subscriptionCountry', '');
    obj.thankYouViewBackgroundImage = _.get(props, 'article.thankYouView.background', '');
    obj.thankYouViewCtaButtonLink = _.get(props, 'article.thankYouView.ctaButtonLink', '');
    obj.thankYouViewCtaButtonText = _.get(props, 'article.thankYouView.ctaButtonText', '');
    obj.thankYouViewDescription = _.get(props, 'article.thankYouView.description', '');
    obj.thankYouViewTitle = _.get(props, 'article.thankYouView.title', '');
    obj.articleTitle = _.get(props, 'article.title', '');

    /* card properties */
    obj.byText = _.get(props, 'byText', '');
    obj.page = _.get(props, 'page', '');
    obj.width = _.get(props, 'width', 1);
}

export function setTitleCardProps(props, obj) {
    /* title card styles */
    obj.backgroundColor = _.get(props, 'styles.backgroundColor', '');
    obj.titleColor = _.get(props, 'styles.titleColor', '');
    obj.descriptionColor = _.get(props, 'styles.descriptionColor', '');
    obj.imageSize = _.get(props, 'styles.imageSize', '');
    obj.verticalAlignment = _.get(props, 'styles.contentVerticalAlignment', '');
    obj.imageHorizontalAlignment = _.get(props, 'styles.imageHorizontalAlignment', '50%');
    obj.imageVerticalAlignment = _.get(props, 'styles.imageVerticalAlignment', '50%');
    obj.customImageSize = _.get(props, 'styles.customImageSize', '');

    /* article properties */
    obj.articleTitle = _.get(props, 'article.title', '');
    obj.articleDescription = _.get(props, 'article.description', '');

    /* image properties */
    obj.iconPath = _.get(props, 'iconPath', '');
    obj.backgroundImageUrl = _.get(props, 'background', '');
    obj.imageAlt = _.get(props, 'imageAlt', '');
    obj.imageSize = obj.imageSize === 'custom' ? obj.customImageSize : obj.imageSize;

    /* card properties */
    obj.width = _.get(props, 'width', 1);
    obj.wide = obj.width === 2 ? 'wide' : '';
}

export function setFeaturedCardProps(props, obj) {
    /* featured card styles */
    obj.backgroundColor = _.get(props, 'styles.backgroundColor', '');
    obj.titleColor = _.get(props, 'styles.titleColor', '');
    obj.descriptionColor = _.get(props, 'styles.descriptionColor', '');
    obj.imageSize = _.get(props, 'styles.imageSize', '');
    obj.contentVerticalAlignment = _.get(props, 'styles.contentVerticalAlignment', '');
    obj.contentHorizontalAlignment = _.get(props, 'styles.contentHorizontalAlignment', '');

    /* article properties */
    obj.articleTitle = _.get(props, 'article.title', '');
    obj.articleDescription = _.get(props, 'article.description', '');
    obj.articleUrl = _.get(props, 'article.url', '');
    obj.ctaButtonLink = _.get(props, 'article.ctaButtonLink', '');
    obj.ctaButtonText = _.get(props, 'article.ctaButtonText', '');

    obj.shouldShowCta = showFeaturedCardCta(props);
    obj.showArticleUrl = shouldShowArticleUrl(props);
    obj.showArticleDescription = shouldShowArticleDescription(props);

    /* image properties */
    obj.backgroundImageUrl = _.get(props, 'background', '');
    obj.imageHorizontalAlignment = _.get(props, 'styles.imageHorizontalAlignment', '50%');
    obj.imageVerticalAlignment = _.get(props, 'styles.imageVerticalAlignment', '50%');

    /* card properties */
    obj.width = _.get(props, 'width', 1);
    obj.wide = obj.width === 2 ? 'wide' : '';
}

export function setStandardCardProps(props, obj) {
    /* article styles */
    obj.backgroundImageUrl = _.get(props, 'background', '');
    obj.backgroundColor = _.get(props, 'styles.backgroundColor', '');
    obj.titleColor = _.get(props, 'styles.titleColor', '');

    /* tag properties */
    obj.tagColor = _.get(props, 'tag.color', '');
    obj.tagUrl = _.get(props, 'tag.url', '');
    obj.tagTitle = _.get(props, 'tag.title', '');
    obj.showTagLabel = shouldShowTagLabel(props);

    /* article properties */
    obj.articleTitle = _.get(props, 'article.title', '');
    obj.articleUrl = _.get(props, 'article.url', '');
    obj.author = _.get(props, 'article.author', '');

    /* video properties */
    obj.videoUrl = _.get(props, 'video.url', '');
    obj.videoPolicy = _.get(props, 'video.policy', '');
    obj.showVideoButton = shouldShowVideoUrl(props);
    obj.pageName = getPageName(obj.articleUrl); // Needed for video modal

    obj.byText = _.get(props, 'byText', '');
    obj.byAuthor = getByAuthorText(obj.author, obj.byText);
}
