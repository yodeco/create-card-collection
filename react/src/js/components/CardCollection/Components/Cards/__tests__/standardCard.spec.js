import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build/index';
import React from 'react';
import toJson from 'enzyme-to-json';
import { setStandardCardProps, getByAuthorText, shouldShowTagLabel, getPageName, shouldShowVideoUrl } from '../../../Utilities/CardUtils';
import StandardCard from '../standardCard';

Enzyme.configure({ adapter: new Adapter() });

describe('Standard Card Tests', () => {
    describe('Whenever the standard card is passed in all empty strings for props', () => {
        it('all props correctly set to the empty string (and false) in the outputted HTML', () => {
            const wrapper = shallow(<StandardCard
                pageName=""
                showVideoButton={false}
                backgroundImageUrl=""
                backgroundColor=""
                titleColor=""
                tagColor=""
                tagUrl=""
                tagTitle=""
                showTagLabel={false}
                articleTitle=""
                articleUrl=""
                videoUrl=""
                videoPolicy=""
                byAuthor="" />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        it('all props correctly set to the empty string (and true) in the outputted HTML', () => {
            const wrapper = shallow(<StandardCard
                pageName=""
                showVideoButton
                backgroundImageUrl=""
                backgroundColor=""
                titleColor=""
                tagColor=""
                tagUrl=""
                tagTitle=""
                showTagLabel
                articleTitle=""
                articleUrl=""
                videoUrl=""
                videoPolicy=""
                byAuthor="" />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('Whenever the standard card is passed in the correct flags for the video button', () => {
        it('The video button will render with the correct props', () => {
            const wrapper = shallow(<StandardCard
                pageName="customer-experience-24-hour-fitness"
                showVideoButton
                backgroundImageUrl=""
                backgroundColor=""
                titleColor=""
                tagColor=""
                tagUrl=""
                tagTitle=""
                showTagLabel={false}
                articleTitle=""
                articleUrl=""
                videoUrl="https://video.tv.adobe.com/v/22288?autoplay=true&quality=9"
                videoPolicy="autoplay; fullscreen"
                byAuthor="" />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('Whenever the standard card is passed in the correct flags for the tag label to show', () => {
        it('The tag label button will render with the correct props', () => {
            const wrapper = shallow(<StandardCard
                pageName=""
                showVideoButton={false}
                backgroundImageUrl=""
                backgroundColor=""
                titleColor=""
                tagColor="#bb121a"
                tagUrl="http://localhost:4502/content/www/us/en/insights/media-entertainment.html"
                tagTitle="Media & Entertainment"
                showTagLabel
                articleTitle=""
                articleUrl=""
                videoUrl=""
                videoPolicy=""
                byAuthor="" />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('Getting Standard Card Props from Card collection', () => {
        describe('When the card collection passes an empty object', () => {
            it('The page name is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.pageName).toBe('');
            });

            it('The background image url is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.backgroundImageUrl).toBe('');
            });


            it('The background color is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.backgroundColor).toBe('');
            });

            it('The title color is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.titleColor).toBe('');
            });

            it('The tag color is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.tagColor).toBe('');
            });

            it('The tag url is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.tagUrl).toBe('');
            });

            it('The tag title is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.tagTitle).toBe('');
            });

            it('The article title is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.articleTitle).toBe('');
            });

            it('The article url is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.articleUrl).toBe('');
            });

            it('The video url is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.videoUrl).toBe('');
            });

            it('The video policy is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.videoPolicy).toBe('');
            });

            it('The by author is defaulted to the empty string', () => {
                const cardCollectionPropsObj = {};
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.byAuthor).toBe('');
            });
        });

        describe('When the card collection passes a byText prop', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = { byText: 'by {authorName}' };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.byText).toBe('by {authorName}');
            });
        });
    });

    describe('Getting article properties', () => {
        describe('When the card collection passes a article Url', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    article: {
                        url: 'http://localhost:4502/content/www/us/en/insights/5-digital-trends-for-financial-marketers.html',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.articleUrl).toBe('http://localhost:4502/content/www/us/en/insights/5-digital-trends-for-financial-marketers.html');
            });
        });

        describe('When the card collection passes a article title', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    article: {
                        title: '5 Digital Trends Financial Marketers Can’t Afford To Ignore',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.articleTitle).toBe('5 Digital Trends Financial Marketers Can’t Afford To Ignore');
            });
        });

        describe('When the card collection passes a article author', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    article: {
                        author: 'John Doe',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.author).toBe('John Doe');
            });
        });
    });

    describe('Getting tag properties', () => {
        describe('When the card collection passes a tag color', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    tag: {
                        color: '#ffffff',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.tagColor).toBe('#ffffff');
            });
        });

        describe('When the card collection passes a tag title', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    tag: {
                        title: 'Advertising',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.tagTitle).toBe('Advertising');
            });
        });

        describe('When the card collection passes a tag url', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    tag: {
                        url: 'http://localhost:4502/content/www/us/en/insights/advertising.html',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.tagUrl).toBe('http://localhost:4502/content/www/us/en/insights/advertising.html');
            });
        });
    });

    describe('Getting article styles', () => {
        describe('When the card collection passes a background image url', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    background: '/content/dam/www/us/en/insights/cropped_images/The%20number%205%20.png',
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.backgroundImageUrl).toBe('/content/dam/www/us/en/insights/cropped_images/The%20number%205%20.png');
            });
        });

        describe('When the card collection passes a background color', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    styles: {
                        backgroundColor: '#ffffff',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.backgroundColor).toBe('#ffffff');
            });
        });

        describe('When the card collection passes a title color', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    styles: {
                        titleColor: '#ffffff',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.titleColor).toBe('#ffffff');
            });
        });
    });

    describe('Getting video properties', () => {
        describe('When the card collection passes a video url', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    video: {
                        url: 'https://video.tv.adobe.com/v/22288?autoplay=true&quality=9',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.videoUrl).toBe('https://video.tv.adobe.com/v/22288?autoplay=true&quality=9');
            });

            describe('When the card collection passes a video policy', () => {
                it('It is successfully mapped to a standard card prop', () => {
                    const cardCollectionPropsObj = {
                        video: {
                            policy: 'autoplay; fullscreen',
                        },
                    };
                    const standardCardPropsObj = {};

                    setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                    expect(standardCardPropsObj.videoPolicy).toBe('autoplay; fullscreen');
                });
            });
        });

        describe('When the card collection passes a background color', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    styles: {
                        backgroundColor: '#ffffff',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.backgroundColor).toBe('#ffffff');
            });
        });

        describe('When the card collection passes a title color', () => {
            it('It is successfully mapped to a standard card prop', () => {
                const cardCollectionPropsObj = {
                    styles: {
                        titleColor: '#ffffff',
                    },
                };
                const standardCardPropsObj = {};

                setStandardCardProps(cardCollectionPropsObj, standardCardPropsObj);
                expect(standardCardPropsObj.titleColor).toBe('#ffffff');
            });
        });
    });

    describe('Testing Getting Author Text Method', () => {
        describe('When using an English By Text and an Actual Author Name', () => {
            it('The author text is shown correctly', () => {
                const author = 'Giselle Abramovich'; // what the JSON returns
                const byText = 'by {authorName}';
                const authorText = getByAuthorText(author, byText);
                expect(authorText).toBe('by Giselle Abramovich');
            });
        });

        describe('When using an English By Text and "by Adobe"', () => {
            it('The author text is shown correctly', () => {
                const author = ' Adobe'; // what the JSON returns
                const byText = 'by {authorName}';
                const authorText = getByAuthorText(author, byText);
                /* There is CSS on the page that transforms this to upper-case */
                /* The extra space is ignored by HTML */
                expect(authorText).toBe('by  Adobe');
            });
        });

        describe('When using a Japanese By Text and "by Unite Editing Unit"', () => {
            it('The author text is shown correctly', () => {
                const author = 'UNITE 編集部'; // what the JSON returns
                const byText = '作者 {authorName}';
                const authorText = getByAuthorText(author, byText);
                expect(authorText).toBe('作者 UNITE 編集部');
            });
        });
    });

    describe('Testing Should Show Tag Label Method', () => {
        describe('When passed in a valid prop object', () => {
            it('The tag should be shown flag returns true', () => {
                const props = {
                    tag: {
                        title: 'Advertising',
                    },
                };
                const showTagLabel = shouldShowTagLabel(props);
                expect(showTagLabel).toBe(true);
            });
        });

        describe('When passed in a invalid prop object', () => {
            it('The tag should be shown flag returns false', () => {
                const props = {
                    tags: {
                        title: 'Advertising',
                    },
                };
                const showTagLabel = shouldShowTagLabel(props);
                expect(showTagLabel).toBe(false);
            });
        });

        describe('When passed in an empty prop object', () => {
            it('The tag should be shown flag returns false', () => {
                const props = {};
                const showTagLabel = shouldShowTagLabel(props);
                expect(showTagLabel).toBe(false);
            });
        });
    });

    describe('Testing Get Page Name Method', () => {
        describe('When passed in an empty string', () => {
            it('The page name method should return an empty string', () => {
                const articleUrl = '';
                const pageName = getPageName(articleUrl);
                expect(pageName).toBe('');
            });
        });

        describe('When passed in an article URL', () => {
            it('The page name method should return just the page name without any extensions', () => {
                const articleUrl =
                    'http://localhost:4502/content/www/us/en/insights/5-digital-trends-for-financial-marketers.html';
                const pageName = getPageName(articleUrl);
                expect(pageName).toBe('5-digital-trends-for-financial-marketers');
            });
        });

        describe('When passed in a relative path URL', () => {
            it('The page name method should return just the page name without any extensions', () => {
                const articleUrl =
                    '/content/www/us/en/insights/5-digital-trends-for-financial-marketers.html';
                const pageName = getPageName(articleUrl);
                expect(pageName).toBe('5-digital-trends-for-financial-marketers');
            });
        });
    });

    describe('Testing Should Show Video Url Method', () => {
        describe('When passed in a valid prop object', () => {
            it('The tag should be shown flag returns true', () => {
                const props = {
                    video: {
                        url: 'https://video.tv.adobe.com/v/22288?autoplay=true&quality=9',
                    },
                };
                const showVideoUrl = shouldShowVideoUrl(props);
                expect(showVideoUrl).toBe(true);
            });
        });

        describe('Whenever the video.url property is missing', () => {
            it('The tag should be shown flag returns false', () => {
                const props = {
                    video: {
                        policy: 'fullscreen auto',
                    },
                };
                const showVideoUrl = shouldShowVideoUrl(props);
                expect(showVideoUrl).toBe(false);
            });
        });

        describe('When passed in an empty prop object', () => {
            it('The tag should be shown flag returns false', () => {
                const props = {};
                const showVideoUrl = shouldShowVideoUrl(props);
                expect(showVideoUrl).toBe(false);
            });
        });
    });
});
