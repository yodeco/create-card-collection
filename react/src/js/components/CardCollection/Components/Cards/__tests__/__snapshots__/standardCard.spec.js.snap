// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standard Card Tests Whenever the standard card is passed in all empty strings for props all props correctly set to the empty string (and false) in the outputted HTML 1`] = `
<div
  className="card card_standard"
>
  <div
    className="card_img"
    style={
      Object {
        "backgroundImage": "url()",
        "backgroundSize": "cover",
      }
    }
  >
    <a
      className="card_hero"
      href=""
    >
      <If
        condition={false}
      >
        <Then>
          <videoButton
            name=""
            videoPolicy=""
            videoUrl=""
          />
        </Then>
      </If>
    </a>
    <If
      condition={false}
    >
      <Then>
        <div
          className="tag_label"
          style={
            Object {
              "backgroundColor": "",
            }
          }
        >
          <a
            href=""
          />
        </div>
      </Then>
    </If>
  </div>
  <div
    className="card_bg"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <h2
      className="card_title"
      style={
        Object {
          "color": "",
        }
      }
    >
      <a
        href=""
      />
    </h2>
    <h4
      className="card_author"
    />
  </div>
</div>
`;

exports[`Standard Card Tests Whenever the standard card is passed in all empty strings for props all props correctly set to the empty string (and true) in the outputted HTML 1`] = `
<div
  className="card card_standard"
>
  <div
    className="card_img"
    style={
      Object {
        "backgroundImage": "url()",
        "backgroundSize": "cover",
      }
    }
  >
    <a
      className="card_hero"
      href=""
    >
      <If
        condition={true}
      >
        <Then>
          <videoButton
            name=""
            videoPolicy=""
            videoUrl=""
          />
        </Then>
      </If>
    </a>
    <If
      condition={true}
    >
      <Then>
        <div
          className="tag_label"
          style={
            Object {
              "backgroundColor": "",
            }
          }
        >
          <a
            href=""
          />
        </div>
      </Then>
    </If>
  </div>
  <div
    className="card_bg"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <h2
      className="card_title"
      style={
        Object {
          "color": "",
        }
      }
    >
      <a
        href=""
      />
    </h2>
    <h4
      className="card_author"
    />
  </div>
</div>
`;

exports[`Standard Card Tests Whenever the standard card is passed in the correct flags for the tag label to show The tag label button will render with the correct props 1`] = `
<div
  className="card card_standard"
>
  <div
    className="card_img"
    style={
      Object {
        "backgroundImage": "url()",
        "backgroundSize": "cover",
      }
    }
  >
    <a
      className="card_hero"
      href=""
    >
      <If
        condition={false}
      >
        <Then>
          <videoButton
            name=""
            videoPolicy=""
            videoUrl=""
          />
        </Then>
      </If>
    </a>
    <If
      condition={true}
    >
      <Then>
        <div
          className="tag_label"
          style={
            Object {
              "backgroundColor": "#bb121a",
            }
          }
        >
          <a
            href="http://localhost:4502/content/www/us/en/insights/media-entertainment.html"
          >
            Media & Entertainment
          </a>
        </div>
      </Then>
    </If>
  </div>
  <div
    className="card_bg"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <h2
      className="card_title"
      style={
        Object {
          "color": "",
        }
      }
    >
      <a
        href=""
      />
    </h2>
    <h4
      className="card_author"
    />
  </div>
</div>
`;

exports[`Standard Card Tests Whenever the standard card is passed in the correct flags for the video button The video button will render with the correct props 1`] = `
<div
  className="card card_standard"
>
  <div
    className="card_img"
    style={
      Object {
        "backgroundImage": "url()",
        "backgroundSize": "cover",
      }
    }
  >
    <a
      className="card_hero"
      href=""
    >
      <If
        condition={true}
      >
        <Then>
          <videoButton
            name="customer-experience-24-hour-fitness"
            videoPolicy="autoplay; fullscreen"
            videoUrl="https://video.tv.adobe.com/v/22288?autoplay=true&quality=9"
          />
        </Then>
      </If>
    </a>
    <If
      condition={false}
    >
      <Then>
        <div
          className="tag_label"
          style={
            Object {
              "backgroundColor": "",
            }
          }
        >
          <a
            href=""
          />
        </div>
      </Then>
    </If>
  </div>
  <div
    className="card_bg"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  >
    <h2
      className="card_title"
      style={
        Object {
          "color": "",
        }
      }
    >
      <a
        href=""
      />
    </h2>
    <h4
      className="card_author"
    />
  </div>
</div>
`;
